x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/actions_menu.test.tsx

```diff
@@ -42,7 +42,7 @@ describe('When using the Endpoint Details Actions Menu', () => {
   let coreStart: AppContextTestRender['coreStart'];
   let renderResult: ReturnType<AppContextTestRender['render']>;
   let httpMocks: ReturnType<typeof endpointPageHttpMock>;
-  let middlewareSpy: AppContextTestRender['middlewareSpy'];
+  // let middlewareSpy: AppContextTestRender['middlewareSpy'];
```

Comment:
I just tried again and I wasn't able to solve this unfortunately. With `userEvent` now being async there might now be a chicken/egg problem with the way this mock is set up with the promise to wait for the middleware event. I suspect it might already have triggered when we try to wait for it. With regular mocks we could do `shouldHaveBeenCalled` after it was called, but I wasn't able to get it to work with this custom approach. I added in some more comments that reference this PR in a80311702f5c. Maybe you can solve it in a follow up? In this case I didn't skip the whole test but just commented the middleware assertion. The rest of the test seems to pass.

Reply:
Hmmm. if the tests are passing, then maybe we no longer need it.
Its been a long time, but I seem to remember creating this middleware "spy" to ensure that we did not do test assertion too soon (before the middleware was able to do what it needed to do). It address test flakiness mostly.

 I'll note this in the internal issue I created so we can revisit... but maybe we no longer need it.